src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/tree/TreeLeaf.java

```diff
@@ -0,0 +1,109 @@
+package de.terrestris.shogun2.model.tree;
+
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIdentityReference;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+
+import de.terrestris.shogun2.model.layer.AbstractLayer;
+
+/**
+ * The TreeLeaf extends the TreeNode and has an associated layer.
+ *
+ * @author Kai Volland
+ * @author Daniel Koch
+ *
+ */
+@Entity
+public class TreeLeaf extends TreeNode {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * 
+	 */
+	@ManyToOne
+	@JsonIdentityInfo(
+			generator=ObjectIdGenerators.PropertyGenerator.class,
+			property="id"
+	)
+	@JsonIdentityReference(alwaysAsId=true)
+	private AbstractLayer layer;
```

Comment:
The leaf should have a real `de.terrestris.shogun2.model.layer.Layer` i think. This would reflect the behaviour of geoext3.